lib/ocpp/v201/charge_point.cpp

```diff
@@ -3077,6 +3096,28 @@ void ChargePoint::handle_heartbeat_response(CallResult<HeartbeatResponse> call)
     }
 }
 
+void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest> call) {
+    EVLOG_debug << "Received SetChargingProfileRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    auto msg = call.msg;
+    SetChargingProfileResponse response;
+    response.status = ChargingProfileStatusEnum::Rejected;
+
+    auto res = this->smart_charging_handler->validate_profile(msg.chargingProfile, msg.evseId);
+    if (res == ProfileValidationResultEnum::Valid) {
+        response = this->smart_charging_handler->add_profile(msg.evseId, msg.chargingProfile);
+        this->callbacks.set_charging_profiles_callback();
+    } else {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = conversions::profile_validation_result_to_reason_code(res);
+        response.statusInfo->additionalInfo = conversions::profile_validation_result_to_string(res);
+        EVLOG_debug << "Rejecting SetChargingProfileRequest:\n reasonCode: " << response.statusInfo->reasonCode.get()
+                    << "\nadditionalInfo: " << response.statusInfo->additionalInfo->get();
```

Comment:
The smallest of nits: I would add a complementary `EVLOG_debug` statement in the other branch of the conditional that reports that the `SetChargingProfileRequest` was accepted.